scripts/get_nbd_extents.py

```diff
@@ -0,0 +1,122 @@
+#!/usr/bin/python
+
+"""
+Returns a list of extents with their block statuses for an NBD export.
+
+This program uses new NBD capabilities introduced in QEMU 2.12.
+
+It uses the BLOCK_STATUS NBD extension, which is documented here:
+https://github.com/NetworkBlockDevice/nbd/blob/extension-blockstatus/doc/proto.md
+The structured replies functionality that this extension relies on is in the
+main branch of the NBD protocol docs:
+https://github.com/NetworkBlockDevice/nbd/blob/master/doc/proto.md
+"""
+
+import argparse
+import json
+import logging
+import logging.handlers
+
+from python_nbd_client import PythonNbdClient
+import python_nbd_client
+
+
+LOGGER = logging.getLogger("get_nbd_extents")
+LOGGER.setLevel(logging.DEBUG)
+# The log level of python_nbd_client is not set, therefore it will default to
+# that of the root logger, which is WARNING by default.
+
+# Request length is a 32-bit int.
+# It looks like for qemu 2.12, the length parameter of a NBD_CMD_BLOCK_STATUS
+# request is not limited by the maximum block size supported by the server (as
+# defined by NBD_INFO_BLOCK_SIZE), only by the size of a 32-bit int.
+MAX_REQUEST_LEN = 2 ** 32 - 1
```

Comment:
Are you sure you want to go up to a 4Gb request?

Reply:
This is for querying the metadata, we're not downloading the actual data.